sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp

```diff
@@ -714,6 +714,47 @@ __esimd_dpasw_nosrc0(__ESIMD_DNS::vector_type_t<T1, N1> src1,
 }
 #endif // !__SYCL_DEVICE_ONLY__
 
+template <typename T, int N>
+__ESIMD_INTRIN std::pair<__ESIMD_DNS::vector_type_t<T, N>,
+                         __ESIMD_DNS::vector_type_t<T, N>>
+__esimd_addc(__ESIMD_DNS::vector_type_t<T, N> src0,
+             __ESIMD_DNS::vector_type_t<T, N> src1)
+#ifdef __SYCL_DEVICE_ONLY__
+    ;
+#else  // !__SYCL_DEVICE_ONLY__
+{
+  __ESIMD_NS::simd<uint64_t, N> Result64 = __ESIMD_NS::simd<T, N>(src0);
+  Result64 += __ESIMD_NS::simd<T, N>(src1);
+  auto Result32 = Result64.template bit_cast_view<T>();
+  __ESIMD_NS::simd<uint32_t, N> CarryV = Result32.template select<N, 2>(1);
+  __ESIMD_NS::simd<uint32_t, N> ResV = Result32.template select<N, 2>(0);
```

Comment:
just for my own understanding, are we setting the stride to 2 here because `Result32` is a bitcasted version of the 64-bit addition, and we want to extract the 32-bit result so we skip 32-bits between every element?